WeatherApp/pabitra0011/js/weather.js

```diff
@@ -0,0 +1,295 @@
+//  ====================== steps or process ===========================
+
+// 1.get location cordinates:: if user input city convert it coordinate and if user use current location we directly get cordinate 
+// 2.when we got coordinate , get weather details form this coordinates..
+// weather ditails is given 3hr gap 5 days 
+// 3. filter these weather and get one forcast per day.
+//4. creates weather cards using these filterd data.
+// 5.  
+
+// =======================================================================
+
+// =======   navbar ==================================================
+const menuIcon = document.querySelector(".humberg-menu i")
+const menuContent = document.querySelector(".menu-content")
+
+menuIcon.addEventListener("click", () => {
+    menuContent.classList.toggle("open")
+    menuIcon.classList.toggle("uil-times")
+})
+// =======================================================================
+// ================= swpaer js ============================
+var swiper = new Swiper(".mySwiper", {
+    slidesPerView: 4,
+    spaceBetween: 10,
+    // loop: true,
+    centeredSlides: true,
+    fade: 'true',
+    spaceBetween: 30,
+    grabCursor: true,
+
+    pagination: {
+        el: ".swiper-pagination",
+        clickable: true,
+        dynamicBullets: true,
+    },
+
+    breakpoints: {
+        0: {
+            slidesPerView: 1,
+        },
+        520: {
+            slidesPerView: 2,
+        },
+        950: {
+            slidesPerView: 3,
+        },
+        1200: {
+            slidesPerView: 4,
+        }
+    },
+});
+
+
+
+// selection all variables ==================================================
+const API_KEY = "28c2b467e5bc6dee079a32184988da3c"
+
+const searchBtn = document.querySelector(".city-submit")
+const input = document.querySelector(".input-city")
+const inputLocation = document.querySelector(".input-location")
+
+const currentWeatherCard = document.querySelector(".current-weather-card");
+const weaklyWeatherCards = document.getElementById("weakly-weather-container")
+const hourlyCards = document.getElementById("hourly-section");
+const swiperSlidCard = document.querySelector(".swiper-slide");
+
```

Comment:
Suppose if some developer in your team wanted to know what these variables are being selected for? Then they would have to look for the corresponding html element in your "index.html" element.

That would create a split attention disorder for your team members. Sorry to put it this way, but yeah if You create a file, it should definitely be a monolithic structure. I know explaining each variable could be harder.

I am just trying to review pull requests on this repo, and kind of like trying to provide a collaborative perspective. That if suppose you were to work in a team then how would your code affect your performance. So nothing personal mate, and well done for your efforts.

Reply:
Thank You @PBJI sir, for spending your precious time to review my PR.

I am new to development, so I don't have much knowledge of structured development. However, I am actively working on improving my skills. Could you please provide some code structure for these lines? It would be very helpful to me.

Thank You again!

Reply:
@pabitra0011 Yes, why not? I will try to refactor your code just after I finish my work. You wouldn't be disappointed I bet.

And thanks for giving designation but I myself learning a lot and the process of learning seems never ending. #KeepLearning

Reply:
Ok